src/test/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/controller/RecallControllerTest.kt

```diff
@@ -394,9 +397,35 @@ class RecallControllerTest {
       equalTo(
         recallResponse.copy(
           documents = listOf(
-            Api.RecallDocument(partADoc2.id(), partADoc2.category, partADoc2.fileName, 2, now),
-            Api.RecallDocument(otherDoc1.id(), otherDoc1.category, otherDoc1.fileName, null, now),
-            Api.RecallDocument(otherDoc2.id(), otherDoc2.category, otherDoc2.fileName, null, now),
+            Api.RecallDocument(partADoc2.id(), partADoc2.category, partADoc2.fileName, 2, now, null),
+            Api.RecallDocument(otherDoc1.id(), otherDoc1.category, otherDoc1.fileName, null, now, null),
+            Api.RecallDocument(otherDoc2.id(), otherDoc2.category, otherDoc2.fileName, null, now, null),
+          )
+        )
+      )
+    )
+  }
+
+  @Test
+  fun `latestDocuments contains the latest details`() {
+    val details2 = " First details"
+    val details3 = " Second details"
+    val partADoc1 =
+      Document(::DocumentId.random(), recallId, PART_A_RECALL_REPORT, "part_a.pdf", 1, OffsetDateTime.now(), null)
+    val partADoc2 = Document(::DocumentId.random(), recallId, PART_A_RECALL_REPORT, "part_a.pdf", 2, now, details2)
+    val partADoc3 = Document(::DocumentId.random(), recallId, PART_A_RECALL_REPORT, "part_a.pdf", 3, now, details3)
+    val recallWithDocuments = recall.copy(documents = setOf(partADoc1, partADoc2, partADoc3))
+
+    every { recallRepository.getByRecallId(recallId) } returns recallWithDocuments
+
+    val response = underTest.getRecall(recallId)
+
+    assertThat(
+      response,
+      equalTo(
+        recallResponse.copy(
```

Comment:
Not sure this makes sense? Since it implies the details are versioned separately to the details. But if you get the latest doc, you are going to get the latest details, I think?

Reply:
So, in the story it says that after 3+ version we are not able to see details of previous version. So, I tried to represent that 3rd version returns details from 3rd uppload